Fix a -Werror=format-security error in R-devel with GCC 15/16, where
`ErrMsg` is passed to `error()` as the format string.

https://github.com/wch/r-source/blob/trunk/src/main/array.c#L302

--- a/src/main/array.c
+++ b/src/main/array.c
@@ -299,7 +299,7 @@
 #else
     if (dn > INT_MAX)
 #endif
-	error(ErrMsg);
+	error("%s", ErrMsg);
     return (R_xlen_t) dn;
 }

